src/components/tutorial.jsx

```diff
@@ -94,26 +94,25 @@ Tutorial.checkIfCompleted = (tutorial, user, preferences) => {
     if (preferences && preferences.preferences && preferences.preferences.tutorials_completed_at && preferences.preferences.tutorials_completed_at[tutorial.id]) {
       return Promise.resolve(preferences.preferences.tutorials_completed_at[tutorial.id]);
     }
+    return Promise.resolve();
   } else {
-    if (completedThisSession[tutorial.id]) {
-      return Promise.resolve(completedThisSession[tutorial.id]);
-    }
+    return Promise.resolve(completedThisSession[tutorial.id]);
```

Comment:
Why was the conditional removed?

Reply:
Because if the conditional is not met (because `completedThisSession[tutorial.id]` doesn't exist) then the promise will not resolve and line 105/6 will return undefined at `.then((completed)`, causing `.then` to fail and the code to stop at that line.

Reply:
Here's how it's currently written in PFE w/ Coffeescript https://github.com/zooniverse/Panoptes-Front-End/blob/2e7e67b553940c0a618163d331912840928ca395/app/components/tutorial.cjsx#L39

Reply:
Ah, ok. In the coffeescript, the existential operator wasn't being used as it is typically, but rather explicitly cast the value of the `completedThisSession[tutorial.id]` to a boolean. We want the Promise to resolve as `true` or `false`. So I believe the conversion of this line to javascript would be:
`return Promise.resolve(!!completedThisSession[tutorial.id]);`

Reply:
Also, line 95:
`return Promise.resolve(!!preferences.preferences.tutorials_completed_at[tutorial.id]);`

Reply:
Sorry scratch that, as there are some existential checks in the original for that one too. But I think you understand what needs to change for both. Thanks.

Reply:
Ahh, interesting. Ok, looks like I'll need to check out how both these promises resolve. Luckily I have a branch set up on BPL, so I can link the two and make sure any changes work correctly when I'm tinkering.